src/@context/UserPreferences.tsx

```diff
@@ -40,97 +45,161 @@ interface UserPreferencesValue {
-    localStorage?.showOnboardingModule ?? showOnboardingModuleByDefault
+    (typeof getCookieValue('onboardingModule') !== 'undefined' &&
+      JSON.parse(getCookieValue('onboardingModule'))?.showOnboardingModule ===
+        'true') ??
+      showOnboardingModuleByDefault
   )
 
   const [onboardingStep, setOnboardingStep] = useState<number>(
-    localStorage?.onboardingStep || 0
+    typeof getCookieValue('onboardingModule') !== 'undefined'
+      ? ~~JSON.parse(getCookieValue('onboardingModule'))?.onboardingStep
+      : 0
   )
+  const twoMonthsFromNow = new Date()
+  twoMonthsFromNow.setMonth(twoMonthsFromNow.getMonth() + 2)
 
   // Write values to localStorage on change
   useEffect(() => {
-    setLocalStorage({
-      chainIds,
-      debug,
-      currency,
-      bookmarks,
-      privacyPolicySlug,
-      showPPC,
-      allowExternalContent,
-      automationWalletJSON: automationWallet,
-      automationWalletMode,
-      showOnboardingModule,
-      onboardingStep
-    })
-  }, [
-    chainIds,
-    debug,
-    currency,
-    bookmarks,
-    privacyPolicySlug,
-    showPPC,
-    allowExternalContent,
-    automationWallet,
-    automationWalletMode,
-    showOnboardingModule,
-    onboardingStep
-  ])
+    if (debug) {
+      setCookie('debug', debug, {
+        expires: twoMonthsFromNow,
+        sameSite: SAME_SITE_OPTIONS.STRICT
+      })
+    } else {
+      deleteCookie('debug')
+    }
+  }, [debug])
+
+  useEffect(() => {
+    if (bookmarks.length > 0) {
+      setCookie('bookmarks', bookmarks.toString())
+    } else {
+      deleteCookie('bookmarks')
+    }
+  }, [bookmarks])
+
+  useEffect(() => {
+    if (chainIds.toString() !== appConfig.chainIds.toString()) {
+      setCookie('chainIds', chainIds.toString())
+    } else {
+      deleteCookie('chainIds')
+    }
+  }, [chainIds])
+
+  useEffect(() => {
+    if (privacyPolicySlug !== defaultPrivacyPolicySlug) {
+      setCookie('privacyPolicySlug', privacyPolicySlug, {
+        expires: twoMonthsFromNow,
+        sameSite: SAME_SITE_OPTIONS.STRICT
+      })
+    } else {
+      deleteCookie('privacyPolicySlug')
+    }
+  }, [privacyPolicySlug])
+
+  useEffect(() => {
+    if (showPPC === false) {
+      setCookie('showPPC', showPPC)
+    } else {
+      deleteCookie('showPPC')
+    }
+  }, [showPPC])
+
+  useEffect(() => {
+    if (allowExternalContent) {
+      setCookie('allowExternalContent', allowExternalContent, {
+        expires: twoMonthsFromNow,
+        sameSite: SAME_SITE_OPTIONS.STRICT
+      })
+    } else {
+      deleteCookie('allowExternalContent')
+    }
+  }, [allowExternalContent])
+
+  useEffect(() => {
+    if (
+      automationWallet !== '' ||
+      automationWalletMode !== AUTOMATION_MODES.SIMPLE
+    ) {
+      const automationWalletCookie = {
+        automationWalletJSON: automationWallet,
+        automationWalletMode
+      }
+      setCookie('automationWallet', JSON.stringify(automationWalletCookie), {
+        expires: null,
```

Comment:
suggestion: add a comment to explain why this is set to `null`

Reply:
additionally, this could be part of the cookies utils constant so that it can be imported where needed